src/System.Management.Automation/engine/SessionStateNavigation.cs

```diff
@@ -258,30 +256,25 @@ private string RemoveQualifier(string path, ProviderInfo provider, out string qu
             {
                 isProviderQualified = true;
 
-                int index = path.IndexOf("::", StringComparison.Ordinal);
-
-                if (index != -1)
-                {
-                    // remove the qualifier
-                    result = path.Substring(index + 2);
-                }
+                // remove the qualifier with '::' trail
+                result = path.Slice(qualifier.Length + 2);
             }
             else
             {
-                if (Globber.IsAbsolutePath(path, out qualifier))
+                if (globber.IsAbsolutePath(path, out string tempQualifier))
                 {
                     isDriveQualified = true;
+                    qualifier = tempQualifier.AsSpan();
```

Comment:
This line doesn't appear to be needed; `tempQualifier` is used below, so this value is not being used, right?

Reply:
It is "out parameter" - we return the value from the method. :-)